pkg/apis/openebs.io/v1alpha1/cstorvolume_policy.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// CStorVolumePolicy describes a configuration required for cstor volume
+// resources
+type CStorVolumePolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines a configuration info of a cstor volume required
+	// to provisione cstor volume resources
+	Spec   CStorVolumePolicySpec   `json:"spec"`
+	Status CStorVolumePolicyStatus `json:"status"`
+}
+
+// CStorVolumePolicySpec ...
+type CStorVolumePolicySpec struct {
+	// TargetSpec represents configuration related to cstor target and its resources
+	Target TargetSpec `json:"target"`
+	// ReplicaSpec represents configuration related to replicas resources
+	Replica ReplicaSpec `json:"replica"`
+}
+
+// TargetSpec represents configuration related to cstor target and its resources
+type TargetSpec struct {
+	// QueueDepth sets the queue size at iSCSI target which limits the
+	// ongoing IO count from client
+	QueueDepth string `json:"queueDepth,omitempty"`
+	// Luworkers sets the number of threads that are working on above queue
+	LuWorkers int64 `json:"luWorkers,omitempty"`
+	// Monitor enables or disables the target exporter sidecar
+	Monitor bool `json:"monitor,omitempty"`
+	// ReplicationFactor represents maximum number of replicas
+	// that are allowed to connect to the target
+	ReplicationFactor int64 `json:"replicationFactor,omitempty"`
```

Comment:
We have the same field in CVC also. Is it redundant? Or planned to handle replica scaleup and scale down with CVP?

Reply:
There are some changes has been going on , with respect to changes in schema